uiowa/uiowa
# Initial PDF functionality for Areas of Study.

Resolves #2500 

### ToDo
- [x] Hide View PDF link from anonymous users.
- [x] Get PDF images working.

### Testing
- `composer install`
- `blt sync --site admissions.uiowa.edu`
- `blt frontend`
- `drush @admissions.local uli /areas-of-study/accounting`
- See the `View PDF` link at the bottom.
- Click the `View PDF` and verify it renders without errors.
  - The featured image is hidden but images in text areas should be displaying.
- Log out and go to /areas-of-study/accounting again. See no `View PDF` link at the bottom.
- Go to http://admissions.local.drupal.uiowa.edu/print/pdf/node/761/ and verify access denied.

### Advanced
- Make a SCSS change in admissions_core/sass/pdf.scss
- Rebuild the workspace: `yarn workspace admissions_core run build`
  - Or watch for changes, see https://github.com/uiowa/uiowa/blob/main/README.md#workspaces
- Rebuild the cache: `drush @admission.local cr`
  - This is needed after any CSS change which is annoying. I'm not sure what cache bin to clear yet. 
- Append `/debug` to the print PDF URL of an AOS. This is what the library uses to generate and you can inspect styles/markup here.

I think most of the styling for this can happen in #2563 but I'm looking for initial thoughts on how this will work. Apparently, PHP PDF libraries have limited CSS support. The one we're using here, dompdf, has limited CSS 3 selector/property support. None seem to have flexbox or grid support. Because of this, we can't just import many of the UIDS component partials. Do we need to? Can we just write very basic styles for these PDFs?
